File: Iterator/Python/ListaPalabras.py
from ListaPalabrasIterador import *

class ListaPalabras:
    
    def __init__(self):
        self._posicion = 0
        self._palabra1 = '' 
        self._palabra2 = '' 
        self._palabra3 = '' 
        self._palabra4 = ''
        self._palabra5 = '' 
        self._posicion = 0
    
    def add(self, p):

        if self._posicion == 0:
            self._palabra1 = p

        if self._posicion == 1:
            self._palabra2 = p
        
        if self._posicion == 2:
            self._palabra3 = p
        
        if self._posicion == 3:
            self._palabra4 = p
        
        if self._posicion == 4:
            self._palabra5 = p
        
        self._posicion += 1
    
    def iterador(self):
        return ListaPalabrasIterador(self._palabra1, self._palabra2, self._palabra3, self._palabra4, self._palabra5)
        

 

File: Iterator/Java/ListaPalabras.java

public class ListaPalabras {
    private String palabra1, palabra2, palabra3, palabra4, palabra5;
    private int posicion;

    ListaPalabras()
    {
        this.posicion = 0;
    }

    public void add(String p)
    {
        switch(this.posicion)
        {
            case 0:
                this.palabra1 = p;
                break;
            case 1:
                this.palabra2 = p;
                break;
            case 2:
                this.palabra3 = p;
                break;
            case 3:
                this.palabra4 = p;
                break;
            case 4:
                this.palabra5 = p;
                break;
        }
        this.posicion++;

    }

    public ListaPalabrasIterador iterador()
    {
        return new ListaPalabrasIterador(this.palabra1, this.palabra2, this.palabra3, this.palabra4, this.palabra5);
    }
}
